### DISCUSSION TITLE
[FeatureRequest] Annotations: Show preview of values when running a query

### DISCUSSION THREAD

**Discussion by jessover9000** [grafana/grafana#38362] [Dashboards, Panels & Visualizations]
User feedback from #grafana-fails / publick slack:
"I found the feedback loop for annotation queries/variable queries to be a bit painful…
There’s not really any feedback on their respective settings pages, so I found myself doing a whole lot of update, go back to dashboard, refresh, see if it worked, repeat.

In an ideal scenario the feedback would be two things (at least for Prometheus data source):
- autocomplete within the Query box
- an on-page display showing the variable values returned by that query"

https://grafana.slack.com/archives/CSKQV9Y5B/p1578923212006800

**Comment by hugohaggmark** [comment]
> User feedback from #grafana-fails / publick slack:
> "I found the feedback loop for annotation queries/variable queries to be a bit painful…

I think this is still true for annotations, not true for Variables as we added loading indicators in both dashboard and on the settings page.

> There’s not really any feedback on their respective settings pages, so I found myself doing a whole lot of update, go back to dashboard, refresh, see if it worked, repeat.

This is only true for annotations then because Variables have always shown a preview of values. I think I'm going to rename and clear up the description a bit.
![image](https://user-images.githubusercontent.com/562238/99034330-31b4bc00-257d-11eb-9d7e-6307fb9c3b75.png)

> In an ideal scenario the feedback would be two things (at least for Prometheus data source):
> 
> * autocomplete within the Query box
> * an on-page display showing the variable values returned by that query"

Again I don't understand this feedback, is the preview hard to find or is the user using an ancient version of Grafana?
![image](https://user-images.githubusercontent.com/562238/99034330-31b4bc00-257d-11eb-9d7e-6307fb9c3b75.png)


> https://grafana.slack.com/archives/CSKQV9Y5B/p1578923212006800

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
